File: 2367_Number_of_Arithmetic_Triplets.py
class Solution:
    def arithmeticTriplets(self, nums: list[int], diff: int) -> int:
        count = 0

        for num in nums:
            if num + diff in nums and num + 2 * diff in nums:
                count += 1

        return count
    
a = Solution()
print(a.arithmeticTriplets([0, 1, 2, 3, 4], 1))  # Output: 4
print(a.arithmeticTriplets([4, 5, 6, 7, 8], 2))  # Output: 3
print(a.arithmeticTriplets([0, 2, 3, 4, 6], 2))  # Output: 1

File: 2367_Number_of_Arithmetic_Triplets.c
# include<stdio.h>
int arithmeticTriplets(int* nums, int numsSize, int diff) {
    int count = 0;
    
    for(int i = 0; i < numsSize; i++){
        int second = 0, third = 0;

        for(int j = i + 1; j < numsSize; j++){
            if(nums[j] - nums[i] == diff){
                second = 1;
            }
            if(nums[j] - nums[i] == 2 * diff){
                third = 1;
            }

            if(second && third){
                count++;
                break; // No need to check further for this i as all values are unique
            }
        }
    }

    return count;
}